common/djangoapps/student/forms.py

```diff
@@ -275,9 +275,15 @@ def __init__(
                         }
                     )
 
+        extended_profile_field_options = configuration_helpers.get_value('EXTRA_FIELD_OPTIONS', [])
+
         for field in self.extended_profile_fields:
-            if field not in self.fields:
-                self.fields[field] = forms.CharField(required=False)
+            options = [(value, _(value)) for value in extended_profile_field_options.get(field, [])]
+
+            if field not in self.fields and options:
+                self.fields[field] = forms.ChoiceField(required=True, choices=options)
```

Comment:
Will all fields be created as "required"? @andrey-canon

Reply:
Yes, if it is included in `REGISTRATION_EXTRA_FIELDS` as hidden,it will not be required

Reply:
Got it.